### PR TITLE
fix() getElementOffset shadow DOM awareness

### LINKED ISSUE #7979: getElementOffset is broken when used with shadow DOM
## Version
5.2.1

## Description
Inside the function getElementOffset, line 164 is:
`docElem = doc.documentElement;`
it should be:
`docElem = doc.documentElement || doc.host;`

It is similar to this:
[https://github.com/fabricjs/fabric.js/issues/2461](url)

### PR DESCRIPTION
**Pull request by YashasviChaurasia**
close #7979
Signed-off-by: Yashasvi Chaurasia <yashasvi12977@gmail.com>

### CODE REVIEW

**Comment by ShaMan123** [comment]
please checkout the package.lock file and rename the PR to something informative

**Comment by YashasviChaurasia** [comment]
Could you please let me know the issue with package.lock file as i am unable to figure out the issue with it.

**Comment by ShaMan123** [comment]
Not what `checkout` means.
RUN:
`git checkout master -- package-lock.json`
and commit.
Your PR should not contain these changes, they are unrelated

**Comment by ShaMan123** [comment]
not good enough
you need to fetch upstream and run
`git checkout upstream/master -- package-lock.json`

**Comment by ShaMan123** [comment]
You should add a test for this use case under `test/unit/util.js`

**Comment by asturur** [comment]
Hi @YashasviChaurasia , since we changed a lot of files recently, could you also open this PR on the 5.x branch? this seems a fix we need to backport to fabric 5 as well.

**Comment by asturur** [comment]
Please in this pr description add the text `close #issuenumber` so that when this pr is merged, the issue will be closed.

**Comment by YashasviChaurasia** [comment]
Ok, i will close this PR and submit a New PR on 5.x branch

**Comment by asturur** [comment]
no we need both, this is good.

**Comment by YashasviChaurasia** [comment]
@ShaMan123 I am unable to curate and fill in a test for `test/unit/util.js`.
Kindly let me know how to proceed.

File: test/unit/util.js [hunk 1]
```diff
@@ -1243,4 +1243,17 @@
...
+
+  QUnit.test.only('getElementOffset inside iframe', function (assert) {
+    var frame = fabric.document.createElement('iframe');
+    fabric.document.body.appendChild(frame);
+    var iframeDoc = frame.contentDocument || frame.contentWindow.document;
+    var el = fabric.document.createElement('canvas');
+    iframeDoc.body.appendChild(el);
+    frame.style.left = '200px';
+    el.style.left = '200px';
+    assert.equal(fabric.util.getElementOffset(el).left, 400);
```
Comment (ShaMan123) [inline test/unit/util.js:1255]: @asturur @YashasviChaurasia I don't understand what the outcome should be...
See failing tests

**Comment by asturur** [comment]
@YashasviChaurasia can you show us how are you using the shadow dom so we can write a test?

**Pull request closed without merging**